### ISSUE TITLE
Names I purchased are still available for purchase?

### ISSUE THREAD

**Issue by csharpner** [namecoin/namecoin-core#255]
Names I've already registered and purchased no longer show up in my registered names list AND if I enter any of them I've already registered _*and paid for*_ , it doesn't give the the warning that they already exist, yet my coin balance has been reduced due to my purchasing them.

1.  My registered names to show up in my list.
2.  That they would no longer be available for others to purchase.
3.  That if the purchase failed, that my coins would be returned.

I registered names, got coin deducted from my wallet, transactions show complete, names DID show up in my registered names list, but now they don't, and when I try to reregister the names, it does NOT tell me that they're already registered.

In the "New name:" field, enter d/myfavoritename and click "Submit".  I get a dialog asking me, "Are you sure you want to register...", I click "Yes", I'm prompted for config information, which I click cancel on.  The name shows up in my registered names list, my coin is deducted, a transaction shows up, and I move on to the next name to register.

nc0.13.99-name-tab-beta1 (64 bit)

Windows 10 Pro N, latest updates.
CPU: Intel Core i5-4690K 3.5Ghz (quadcore)
16GB RAM
Boot drive = 512GB SSD
Data Drive = 12TB RAID 3 Tower with four 4TB drives. App is installed on that drive.

**Comment by domob1812** [comment]
First of all, would you mind sharing one of the names you are talking about here so that we can look what exactly happened with them?

Trying to reregister names that exist already is possible, although the registration will not go through (stopped in the `name_new` phase).  This is a "feature", although an advanced one - for that reason, it was recently made opt-in.  See #54 for more details.

**Comment by csharpner** [comment]
"tyranny" was one of the names.

**Comment by domob1812** [comment]
@csharpner:  If you mean `d/tyranny`, then the name is expired and should be available for registration.  In case you are not aware of that, names expire after 36,000 blocks unless they are updated in that time frame.  So perhaps that's what happened to you (i.e. you registered the name a long time ago, let it expire, so now it is available for re-registration)?

**Comment by csharpner** [comment]
@domob1812 , no.  I just started getting into namecoin a few days ago.  I registered the name and within hours it showed it was available again.

**Comment by domob1812** [comment]
Then something else must be wrong.  What is the transaction ID (txid) where you tried to register the name?

**Comment by csharpner** [comment]
@domob1812 , impossible to know.  All the names I registered no longer show up.  My manage names list is now empty when it once had dozens in it.  Though, I have transactions in my transactions tab.  No way to tell which domain name they're associated with, that I know of.  Here's the most recent one:
8034732e82c2b39716230f2d5804566f4e0fc2bae6bd10e8551c8f2485e2b44d
and here's the oldest one:
b98229a44850da6f1a13f0ac05b5af13273f72986ca0af08013e910e1313326b

**Comment by domob1812** [comment]
Both of the transactions you list are `name_new`'s, which do not yet fully register the name (this is a two step process, which requires *both* `name_new` and `name_firstupdate`).  So perhaps you just did `name_new`'s without the matching `name_firstupdate`'s?

**Comment by csharpner** [comment]
@domob1812 That's probably it:

"In the "New name:" field, enter d/myfavoritename and click "Submit". I get a dialog asking me, "Are you sure you want to register...", I click "Yes", I'm prompted for config information, which I click cancel on. The name shows up in my registered names list, my coin is deducted, a transaction shows up, and I move on to the next name to register."

I use the UI to do this.  I don't have config information.  How do I register and keep a domain name without having a place to host it yet, preferably with the GUI.  And, how to I add this to names I've already registered and paid for without having to buy them again?  I've got transactions, but no way to know which domain names they're associated with.

BTW, I registered one last night and entered some config information this time and that one is still showing up in my wallet.  So, I suspect it is as you say, that there was no config information added.

Thanks!

**Comment by domob1812** [comment]
For the registration, it doesn't matter what you put into the name - even if you don't have a website for it yet, you can still register it.  Perhaps @JeremyRand or @brandonrobets can tell you how to register a name with an empty value using the UI.

**Comment by JeremyRand** [comment]
(Tagging @brandonrobertz since it looks like @domob1812 misspelled his username.)

So first off, I haven't actually tried to use the GUI to register a name with an empty value... my understanding is that clicking "Cancel" in nc0.13.99-name-tab-beta1 will just use an empty value and should work fine at registering, but I haven't tried this (and I think newer versions of Brandon's GUI no longer have a "Cancel" button there, and instead allow entering an empty string).

If you want to use the nc0.13.99-name-tab-beta1 GUI to register names without clicking "Cancel" and you want to minimize your transaction fees / blockchain storage, just enter a single character of your choice or something like that.

Anyway, to rescue the registrations you've already made, I believe what you can do is this:

1. Look in the `debug.log` file in your Namecoin profile folder; it should include lines for each name_new transaction you've made, which include a long hex string called "txid" or something similar, a shorter hex string called "rand" or "salt", and the name being registered.
2. From the Namecoin Core debug console, for each name_new transaction, enter something like this: `name_firstupdate name salt txid X` substituting the name, salt, and txid from `debug.log` and using whatever value you like in place of `X`.

AFAIK if you mess up the command, you won't lose any coins, it will just error, so don't worry too much about getting it right on the first try.

Let us know if you have trouble with any of this and we can provide more info.  Cheers.

**Comment by csharpner** [comment]
Thanks @JeremyRand !  I do see all of the transactions and names in the debug.log file.  I'll go through and recover them all.
